packages/liveblocks-react/src/liveblocks.tsx

```diff
@@ -990,29 +989,26 @@ function useUserThreads_experimental<M extends BaseMetadata>(
     },
   }
 ): ThreadsAsyncResult<M> {
-  const queryKey = React.useMemo(
-    () => makeUserThreadsQueryKey(options.query),
-    [options]
-  );
-
   const client = useClient<M>();
 
   const { store, incrementUserThreadsQuerySubscribers } =
     getExtrasForClient<M>(client);
 
   useEffect(() => {
-    store.waitUntilUserThreadsLoaded(options, queryKey).catch(() => {
+    store.waitUntilUserThreadsLoaded(options.query).catch(() => {
       // Deliberately catch and ignore any errors here
     });
-  }, [queryKey, store]); // eslint-disable-line react-hooks/exhaustive-deps
+  }, [store, options.query]);
```

Comment:
Only possible issue I can think of with this change is that if the user passes in an inline object to the hook like the following:

```tsx
useUserThreads({ query: { resolved: false } });
```

The code inside `useEffect` will run on each render as the `query` gets recreated on each render.

```tsx
useEffect(() => {
  store.waitUntilUserThreadsLoaded(options.query).catch(() => {
    // Deliberately catch and ignore any errors here
  });
}, [store, options.query]);
```

Because we used `queryKey` (which is a string) instead of `options.query` in the past as dependency (and ignored the dependency rule for `useEffect`), this was not an issue although still a hack.

That being said, I think this is fine because the `store.waitUntilUserThreadsLoaded` function will only make a fetch request once.

Reply:
We discussed this yesterday, and this is indeed a bug I introduced with this refactoring! Great catch, @nimeshnayaju! 🙏